profile_test.go

```diff
@@ -0,0 +1,313 @@
+package eat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	InvalidProfile = 100
+	InvalidURL     = "abcd"
+	EmptyURL       = ""
+	InvalidOID     = "xxxx"
+	EmptyOID       = ""
+)
+
+// TestProfile_GetSet_Basic_URL tests the basic setting of Profile as URL string
+func TestProfile_GetSet_Basic_URL(t *testing.T) {
+	inputURL := "https://samplewebsite.com"
+
+	profile, err := NewProfile(inputURL)
+	assert.Nil(t, err)
+
+	expectedURL, err := profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedURL, inputURL)
+	assert.True(t, profile.IsURI())
+
+	inputURL = "https://samplenewwebsite.co.uk"
+	err = profile.Set(inputURL)
+	assert.Nil(t, err)
+
+	expectedURL, err = profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedURL, inputURL)
+	assert.True(t, profile.IsURI())
+
+	// Negative test cases, below
+	inputURL = InvalidURL
+	err = profile.Set(inputURL)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "profile decode failed no valid URL or OID")
+
+	inputURL = EmptyURL
+	err = profile.Set(inputURL)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "profile decode failed no valid URL or OID")
+}
+
+// TestProfile_GetSet_Basic_OID tests the saving and retrieval of OID as profile
+func TestProfile_GetSet_Basic_OID(t *testing.T) {
+	inputOID := "1.2.3.4"
+	profile, err := NewProfile(inputOID)
+	assert.Nil(t, err)
+
+	expectedOID, err := profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedOID, inputOID)
+	assert.True(t, profile.IsOID())
+
+	inputOID = "24.43.27.88"
+	err = profile.Set(inputOID)
+	assert.Nil(t, err)
+
+	expectedOID, err = profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedOID, inputOID)
+	assert.True(t, profile.IsOID())
+
+	// Negative test cases below
+	// Add a test for OID less than minimum arcs
+	inputOID = "56.78"
+	err = profile.Set(inputOID)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "invalid OID, num arcs: 2 < min OID arcs 3")
+
+	inputOID = InvalidOID
+	err = profile.Set(inputOID)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "failed to extract OID from string")
+
+	inputOID = EmptyOID
+	err = profile.Set(inputOID)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "no valid OID")
+}
+
+// TestProfileURI_MarshalCBOROK tests the CBOR marshaling of profile been set as an URI
```

Comment:
we don't need a comment, the test function name should be expressive enough (also the comment is out-of-sync)